### PR TITLE
[DS-3841] enable CORS header for security controller

### PR DESCRIPTION
**Pull request by lap82**
This PR contains:
- setup current user into context object
- enable and expose the cors header for security controller

### CODE REVIEW

**Comment by tdonohue** [comment]
Please rebase this PR against the latest `master` branch.  The `master` branch now requires adherence to our new Java Code Style documented at: https://wiki.duraspace.org/display/DSPACE/Code+Style+Guide

If you have any questions, feel free to ask in this PR or on our Slack: https://wiki.duraspace.org/display/DSPACE/Slack

**Review by tomdesair — APPROVED** [review]
The changes look logical to me, so I'm good with this.

File: dspace-spring-rest/src/main/java/org/dspace/app/rest/security/StatelessAuthenticationFilter.java [hunk 1]
```diff
@@ -78,6 +78,7 @@ private Authentication getAuthentication(HttpServletRequest request) {
             if (eperson != null) {
                 //Pass the eperson ID to the request service
                 requestService.setCurrentUserId(eperson.getID());
+                context.setCurrentUser(eperson);
```
Comment (tomdesair) [inline dspace-spring-rest/src/main/java/org/dspace/app/rest/security/StatelessAuthenticationFilter.java]: Just a minor remarks, I don't think this is necessary as this is already done in the `EPersonClaimProvider` (https://github.com/DSpace/DSpace/blob/master/dspace-spring-rest/src/main/java/org/dspace/app/rest/security/jwt/EPersonClaimProvider.java#L46)

**Comment by tdonohue** [comment]
Closing in favor of #1983 (rebased version).  The other PR was merged, and it included these commits

**Pull request closed without merging**